src/Integration/LocalServices/IProjectSystemHelperExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="IProjectSystemHelperExtensions.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal static class IProjectSystemHelperExtensions
+    {
+        /// <summary>
+        /// Returns whether or not a project is of a known test project type.
```

Comment:
Good point. There's still some inconsistency around how the various components (SonarLint, SonarAnalyzer, the MSBuild Scanner) detect what is and what isn't a test project.
The analysers also check [Nx]Unit assembly references whereas we do not (nor does the scanner AFAIK).
I think the idea would be that this method should return `true` for all known types (including [Nx]Unit projects), but not at the moment.